janusgraph-core/src/main/java/org/janusgraph/graphdb/database/management/ManagementLogger.java

```diff
@@ -122,38 +124,70 @@ public void sendCacheEviction(Set<JanusGraphSchemaVertex> updatedTypes,
 
-        private EvictionTrigger(long evictionId, List<Callable<Boolean>> updatedTypeTriggers, Set<String> openInstances) {
+        private EvictionTrigger(long evictionId, List<Callable<Boolean>> updatedTypeTriggers, StandardJanusGraph graph) {
+            this.graph = graph;
             this.evictionId = evictionId;
             this.updatedTypeTriggers = updatedTypeTriggers;
-            this.openInstances = ImmutableSet.copyOf(openInstances);
-            this.ackCounter = new AtomicInteger(openInstances.size());
+            final JanusGraphManagement mgmt = graph.openManagement();
+            this.instancesToBeAcknowledged = ConcurrentHashMap.newKeySet();
+            ((ManagementSystem) mgmt).getOpenInstancesInternal().forEach(instancesToBeAcknowledged::add);
+            mgmt.rollback();
         }
 
         void receivedAcknowledgement(String senderId) {
-            if (openInstances.contains(senderId)) {
-                int countdown = ackCounter.decrementAndGet();
+            if (instancesToBeAcknowledged.remove(senderId)) {
+                final int ackCounter = instancesToBeAcknowledged.size();
                 log.debug("Received acknowledgement for eviction [{}] from senderID={} ({} more acks still outstanding)",
-                        evictionId, senderId, countdown);
-                if (countdown==0) { //Trigger actions
-                    for (Callable<Boolean> trigger : updatedTypeTriggers) {
-                        try {
-                            boolean success = trigger.call();
-                            assert success;
-                        } catch (Throwable e) {
-                            log.error("Could not execute trigger ["+trigger.toString()+"] for eviction ["+evictionId+"]",e);
-                        }
-                    }
-                    log.info("Received all acknowledgements for eviction [{}]",evictionId);
-                    evictionTriggerMap.remove(evictionId,this);
+                        evictionId, senderId, ackCounter);
+                if (ackCounter == 0) {
+                    runTriggers();
+                }
+            }
+        }
+
+        void runTriggers() {
+            for (Callable<Boolean> trigger : updatedTypeTriggers) {
+                try {
+                    boolean success = trigger.call();
+                    assert success;
+                } catch (Throwable e) {
+                    log.error("Could not execute trigger ["+trigger.toString()+"] for eviction ["+evictionId+"]",e);
                 }
             }
+            log.info("Received all acknowledgements for eviction [{}]",evictionId);
+            evictionTriggerMap.remove(evictionId,this);
+        }
+
+        int removeDroppedInstances() {
+            final JanusGraphManagement mgmt = graph.openManagement();
+            final Set<String> updatedInstances = ((ManagementSystem) mgmt).getOpenInstancesInternal();
+            instancesToBeAcknowledged.forEach(it -> {
```

Comment:
instancesToBeAcknowledged.stream().filter(it -> !updatedInstances.contains(it)).filter(instancesToBeAcknowledged::remove).forEach(it -> log.debug())